### PR TITLE
Google Workspace Authentication

### PR DESCRIPTION
**Pull request by vakohl**
Required items, please complete
   
   Change(s):
   - Adding New ASIM Parser for Google Workspace Authentication

   Reason for Change(s):
   - New ASIM

   Version Updated:
   - 0.1.0

   Testing Completed:
   - Yes

   Checked that the validations are passing and have addressed any issues that are present:
   - Will do

### CODE REVIEW

File: Parsers/ASimAuthentication/Tests/GoogleWorkspace_ASimAuthentication_DataTest.csv [hunk 1]
```diff
@@ -0,0 +1,11 @@
+﻿Result
+"(0) Error: 1 invalid value(s) (up to 10 listed) in 3 records (0.1%) for field [EventResultDetails] of type [Enumerated]: [""MFA not satisfied""] (Schema:Authentication)"
+"(0) Error: 1 invalid value(s) (up to 10 listed) in 3012 records (100.0%) for field [EventProduct] of type [Enumerated]: [""Workspace""] (Schema:Authentication)"
```
Comment (anki-narravula) [inline Parsers/ASimAuthentication/Tests/GoogleWorkspace_ASimAuthentication_DataTest.csv:3]: @vakohl - Can you please fix these errors

File: Parsers/ASimAuthentication/Tests/GoogleWorkspace_ASimAuthentication_SchemaTest.csv [hunk 2]
```diff
@@ -0,0 +1,94 @@
+﻿Result
+"(0) Error: Missing mandatory field [Type]"
```
Comment (anki-narravula) [inline Parsers/ASimAuthentication/Tests/GoogleWorkspace_ASimAuthentication_SchemaTest.csv:2]: @vakohl can you please fix these errors.

File: Parsers/ASimAuthentication/Parsers/ASimAuthenticationGoogleWorkspace.yaml [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+                        sensitive_action_name_s
+                    ),
+          EventResult = case(
+                    event_name_s in ('login_challenge', 'login_verification') and login_challenge_status_s == "passed",
+                    "Success",
+                    event_name_s in ('login_challenge', 'login_verification') and login_challenge_status_s == "incorrect_answer_entered",
+                    "Failure",
+                    EventResult
+                ),
+          EventResultDetails = iif(event_name_s in ('login_challenge', 'login_verification') and login_challenge_status_s == "incorrect_answer_entered", "MFA not satisfied", ""),
```
Comment (anki-narravula) [inline Parsers/ASimAuthentication/Parsers/ASimAuthenticationGoogleWorkspace.yaml:110]: is Blank accepted for enumeration field, do you mean NA ?

File: Parsers/ASimAuthentication/Parsers/vimAuthenticationGoogleWorkspace.yaml [hunk 4]
```diff
@@ -0,0 +1,175 @@
+Parser:
+  Title: Authentication ASIM filtering parser for <product name>
```
Comment (anki-narravula) [inline Parsers/ASimAuthentication/Parsers/vimAuthenticationGoogleWorkspace.yaml:2]: Please replace product name

**Review by anki-narravula — CHANGES_REQUESTED** [review]

File: Parsers/ASimAuthentication/Tests/GoogleWorkspace_ASimAuthentication_DataTest.csv (hunk 1 above)
Comment (vakohl) [reply Parsers/ASimAuthentication/Tests/GoogleWorkspace_ASimAuthentication_DataTest.csv:3]: @anki-narravula I'm adding these new values in Tester.csv. I think these errors will be fixed once the new values are merged in tester.

File: Parsers/ASimAuthentication/Tests/GoogleWorkspace_ASimAuthentication_SchemaTest.csv (hunk 2 above)
Comment (vakohl) [reply Parsers/ASimAuthentication/Tests/GoogleWorkspace_ASimAuthentication_SchemaTest.csv:2]: @anki-narravula not seeing this error after re-running tests. Updated results in PR

File: Parsers/ASimAuthentication/Parsers/ASimAuthenticationGoogleWorkspace.yaml (hunk 3 above)
Comment (vakohl) [reply Parsers/ASimAuthentication/Parsers/ASimAuthenticationGoogleWorkspace.yaml:110]: @anki-narravula EventResultDetails is not a mandatory field. can be kept as blank. Here, I want to set this value only when event_name_s  is 'login_challenge' or 'login_verification'

File: Parsers/ASimAuthentication/Parsers/vimAuthenticationGoogleWorkspace.yaml (hunk 4 above)
Comment (vakohl) [reply Parsers/ASimAuthentication/Parsers/vimAuthenticationGoogleWorkspace.yaml:2]: thanks @anki-narravula. done.

**Comment by vakohl** [comment]
@anki-narravula I've checked all comments. Please validate.

**Review by anki-narravula — DISMISSED** [review]

**Review by anki-narravula — APPROVED** [review]

**Pull request merged**